From 16485d756682108002276c3f02ee76572d679ffb Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Wed, 8 Jan 2025 17:45:28 +0100
Subject: [PATCH 1430/1695] drivers: stm32_tamp: support masking potential
 reset for STM32MP21

Add support of "st,tamp-mask-pot-reset" property for STM32MP21.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 144b46da1be9860cd89a52f7c338df06deb01fc8 drivers: stm32_tamp: support masking potential reset for STM32MP21]
Change-Id: I7b808600c410eafb83c09aa10ff814c77592704c
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/427742
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/drivers/stm32_tamp.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/core/drivers/stm32_tamp.c b/core/drivers/stm32_tamp.c
index 26b3981ac..03a8362e6 100644
--- a/core/drivers/stm32_tamp.c
+++ b/core/drivers/stm32_tamp.c
@@ -35,9 +35,9 @@
 #include <libfdt.h>
 #include <mm/core_memprot.h>
 #include <stdbool.h>
-#if defined(CFG_STM32MP25) || defined(CFG_STM32MP23)
+#if defined(CFG_STM32MP25) || defined(CFG_STM32MP23) || defined(CFG_STM32MP21)
 #include <stm32_sysconf.h>
-#endif /* defined(CFG_STM32MP25) || defined(CFG_STM32MP23) */
+#endif /* defined(CFG_STM32MP25) || defined(CFG_STM32MP23) || defined(CFG_STM32MP21) */
 #include <stm32_util.h>
 
 
@@ -2068,10 +2068,10 @@ static TEE_Result stm32_tamp_probe(const void *fdt, int node,
 		goto err;
 	}
 
-#if defined(CFG_STM32MP25) || defined(CFG_STM32MP23)
+#if defined(CFG_STM32MP25) || defined(CFG_STM32MP23) || defined(CFG_STM32MP21)
 	if (stm32_tamp.pdata.mask_pot_reset)
 		stm32mp_syscfg_write(SYSCFG_POTTAMPRSTCR, BIT(0), BIT(0));
-#endif /* defined(CFG_STM32MP25) || defined(CFG_STM32MP23) */
+#endif /* defined(CFG_STM32MP25) || defined(CFG_STM32MP23) || defined(CFG_STM32MP21) */
 
 	/*
 	 * Select extra IP to add in the deleted/blocked IP in case of
-- 
2.47.2

